src/components/chart/ha-sankey-chart.ts

```diff
@@ -65,141 +51,144 @@ export class HaSankeyChart extends LitElement {
-                  style="fill: ${node.color}"
-                >
-                  <title>${node.tooltip}</title>
-                </rect>
-                ${
-                  this.vertical
-                    ? nothing
-                    : svg`
-                      <text 
-                        class="node-label" 
-                        x=${NODE_WIDTH + 5}
-                        y=${node.size / 2}
-                        text-anchor="start" 
-                        dominant-baseline="middle"
-                      >${node.label}</text>
-                    `
-                }
-              </g>
-            `
-        )}
-      </svg>
-      ${this.vertical
-        ? nodes.map((node) => {
-            if (!node.label) {
-              return nothing;
+  private _createOptions = memoizeOne(
+    () =>
+      ({
+        grid: {
+          top: 0,
+          bottom: 0,
+          left: 0,
+          right: 0,
+        },
+        tooltip: {
+          trigger: "item",
+          formatter: (params: CallbackDataParams) => {
+            const data = params.data as Record<string, any>;
+            const value = this.valueFormatter
+              ? this.valueFormatter(data.value)
+              : data.value;
+            if (data.id) {
+              const node = this.data.nodes.find((n) => n.id === data.id);
+              return `${params.marker} ${node?.label ?? data.id}<br>${value}`;
             }
-            const labelWidth = MIN_DISTANCE + node.size;
-            const fontSize = this._getVerticalLabelFontSize(
-              node.label,
-              labelWidth
+            if (data.source && data.target) {
+              const source = this.data.nodes.find((n) => n.id === data.source);
+              const target = this.data.nodes.find((n) => n.id === data.target);
+              return `${source?.label ?? data.source} → ${target?.label ?? data.target}<br>${value}`;
+            }
+            return null;
+          },
+          appendTo: document.body,
+        },
+      }) as ECOption
+  );
+
+  private _createData = memoizeOne((data: SankeyChartData, width = 0) => {
+    const filteredNodes = data.nodes.filter((n) => n.value > 0);
+    const indexes = [...new Set(filteredNodes.map((n) => n.index))];
+    const links = this._processLinks(filteredNodes, data.links);
+    const sectionWidth = width / indexes.length;
+    const labelSpace = sectionWidth - NODE_SIZE - LABEL_DISTANCE;
+
+    return {
+      id: "sankey",
+      type: "sankey",
+      nodes: filteredNodes.map((node) => ({
+        id: node.id,
+        value: node.value,
+        itemStyle: {
+          color: node.color,
+        },
+        depth: node.index,
+      })),
+      links,
+      draggable: false,
+      orient: this.vertical ? "vertical" : "horizontal",
+      nodeWidth: 15,
+      nodeGap: NODE_GAP,
+      lineStyle: {
+        color: "gradient",
+        opacity: 0.4,
+      },
+      layoutIterations: 0,
```

Comment:
This just keeps the node locations as they are and looks ok. Increasing it allows echarts to "improve" the layout but in my testing I had to set it to 128 to get an almost good enough layout